addons/event/models/event_mail.py

```diff
@@ -114,13 +119,48 @@ def execute(self):
                 self.write({'mail_sent': True})
         return True
 
+    @api.model
+    def _warn_template_error(self, scheduler, exception):
+        if random.random() < 0.15:
+            ex_s = exception_to_unicode(exception)
+            try:
+                event, template = scheduler.event_id, scheduler.template_id
+                emails = list(set([event.organizer_id.email, event.user_id.email, template.write_uid.email]))
+                subject = _("WARNING: Event Scheduler Error for event: %s" % event.name)
+                body = _("""Event Scheduler for:
+                              - Event: %s (%s)
+                              - Planified: %s
+                              - Template: %s (%s)
+                            \nFailed with error:
+                              - %s
+                            \nYou receive this email because you are:
+                            - the organizer of the event,
+                            - or the responsible of the event,
+                            - or the last writer of the template."""
+                         % (event.name, event.id, scheduler.scheduled_date, template.name, template.id, ex_s))
+                email = self.env['ir.mail_server'].build_email(
+                    email_from=self.env.user.email,
+                    email_to=emails,
+                    subject=subject, body=body,
+                )
+                self.env['ir.mail_server'].send_email(email)
+            except Exception as e:
+                _logger.error("Exception while sending traceback by email: %s.\n Original Traceback:\n%s", e, exception)
+                pass
+
     @api.model
     def run(self, autocommit=False):
         schedulers = self.search([('done', '=', False), ('scheduled_date', '<=', datetime.strftime(fields.datetime.now(), tools.DEFAULT_SERVER_DATETIME_FORMAT))])
         for scheduler in schedulers:
-            scheduler.execute()
-            if autocommit:
-                self.env.cr.commit()
+            try:
+                with self.env.cr.savepoint():
```

Comment:
Hello @JKE-be 

I have been struggling with an issue happening in this method.
When the event_mail scheduler has to send emails *that contain reports*, I get a:
`bad query: RELEASE SAVEPOINT "669516d2915811eba8d2b4d5bde7cdcc"`
Because the savepoint does not exist anymore.

After a long time investigating, I think this is caused by the combination of the savepoint done in your commit here.

And the fact that we "commit the assets bundle" (not exactly sure what that entails) with the context key used during the reporting process:
https://github.com/odoo/odoo/blob/master/odoo/addons/base/models/ir_actions_report.py#L728

Which causes the transaction to be committed here:
https://github.com/odoo/odoo/blob/master/odoo/addons/base/models/assetsbundle.py#L336

Which I believe kills the previously saved savepoint and leads to my issue.

I'm not sure exactly how I could get around that.
I have seen this because I'm trying to add a report to a mail template in one of my tasks ( #64066 ).
But I'm pretty sure this can already happen in production if you setup reminder emails that contain a report.

I'm guessing the savepoint is there for a specific reason.
Do you have more insights on a solution for this?

Thanks for your help.

Reply:
Ping @JKE-be 

When you have some time, can you take a look?
I would like to un-block my task.
Sorry to bother you on this old thing.

Thanks.